### DISCUSSION TITLE
Autokey script run another Autokey script

### DISCUSSION THREAD

**Discussion by ZeRenCZ** [autokey/autokey#771] [Q&A] [answered]
So if I have scripts:

Script 1:
`keyboard.send_key('<f1>')`
`keyboard.send_key('u')`
`Run "Script 2"` 

Script 2:
`keyboard.send_key('<alt>')`

is it possible for script 1 to run script 2?

btw. would be even possible to run bash script?

**Comment by Elliria** [comment, marked as answer]
Yes to both, but you'll want to replace `Run "Script 2"` with `engine.run_script("Script 2")` to use AutoKey's syntax to run the second script.

You can see a demonstration of that in the [Run an AutoKey script from an AutoKey script](https://github.com/autokey/autokey/wiki/Advanced-Scripts#run-an-autokey-script-from-an-autokey-script) section of the AutoKey wiki's [Basic Scripts](https://github.com/autokey/autokey/wiki/Basic-Scripts) page.

Also, the [Start external scripts](https://github.com/autokey/autokey/wiki/Basic-Scripts#start-external-scripts) section of the AutoKey wiki's [Basic Scripts](https://github.com/autokey/autokey/wiki/Basic-Scripts) page demonstrates how to run a Bash script from inside of an AutoKey script.

In case you haven't explored it yet, you can find [the AutoKey wiki here](https://github.com/autokey/autokey/wiki). You may find the [Example Scripts](https://github.com/autokey/autokey/wiki/Example-Scripts) page especially interesting. It's filled with lots of examples and you're welcome to [contribute some of your own](https://github.com/autokey/autokey/wiki/Contribute-to-the-Wiki) if you like.

Last, but not least, I noticed that you're using the `keyboard.send_key()` method in your script. Both that and the `keyboard.send_keys()` method can be used to send a single keystroke, but they behave differently. See [this note in the Key Combinations page of the wiki](https://github.com/autokey/autokey/wiki/Key-Combinations#about-the-keyboardsend_key-and-keyboardsend_keys-methods-and-case) for a description of the difference. Notes about this have also now been added to the **Send key** and **Send keys** sections of the [API Examples](https://github.com/autokey/autokey/wiki/API-Examples) page.
